src/SprykerEco/Zed/Unzer/Business/Credentials/UnzerCredentialsCreator.php

```diff
@@ -0,0 +1,216 @@
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsStoreRelationUpdaterInterface
+     */
+    protected $unzerCredentialsStoreRelationUpdater;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Writer\UnzerVaultWriterInterface
+     */
+    protected $unzerVaultWriter;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Dependency\UnzerToUtilTextServiceInterface
+     */
+    protected $utilTextService;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Notification\Configurator\UnzerNotificationConfiguratorInterface
+     */
+    protected $unzerNotificationConfigurator;
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+     */
+    protected $unzerReader;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Persistence\UnzerEntityManagerInterface $unzerEntityManager
+     * @param \SprykerEco\Zed\Unzer\Business\Credentials\UnzerCredentialsStoreRelationUpdaterInterface $unzerCredentialsStoreRelationUpdater
+     * @param \SprykerEco\Zed\Unzer\Business\Writer\UnzerVaultWriterInterface $unzerVaultWriter
+     * @param \SprykerEco\Zed\Unzer\Dependency\UnzerToUtilTextServiceInterface $utilTextService
+     * @param \SprykerEco\Zed\Unzer\Business\Notification\Configurator\UnzerNotificationConfiguratorInterface $unzerNotificationConfigurator
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     */
+    public function __construct(
+        UnzerEntityManagerInterface $unzerEntityManager,
+        UnzerCredentialsStoreRelationUpdaterInterface $unzerCredentialsStoreRelationUpdater,
+        UnzerVaultWriterInterface $unzerVaultWriter,
+        UnzerToUtilTextServiceInterface $utilTextService,
+        UnzerNotificationConfiguratorInterface $unzerNotificationConfigurator,
+        UnzerReaderInterface $unzerReader
+    ) {
+        $this->unzerEntityManager = $unzerEntityManager;
+        $this->unzerCredentialsStoreRelationUpdater = $unzerCredentialsStoreRelationUpdater;
+        $this->unzerVaultWriter = $unzerVaultWriter;
+        $this->utilTextService = $utilTextService;
+        $this->unzerNotificationConfigurator = $unzerNotificationConfigurator;
+        $this->unzerReader = $unzerReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerCredentialsTransfer $unzerCredentialsTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsResponseTransfer
+     */
+    public function createUnzerCredentials(UnzerCredentialsTransfer $unzerCredentialsTransfer): UnzerCredentialsResponseTransfer
+    {
+        $propelConnection = Propel::getConnection();
+        $unzerCredentialsResponseTransfer = new UnzerCredentialsResponseTransfer();
+        $propelConnection->beginTransaction();
+
+        try {
+            $unzerCredentialsResponseTransfer = $this->executeCreateUnzerCredentialsTransaction($unzerCredentialsTransfer);
+            $propelConnection->commit();
+        } catch (UnzerApiException $unzerApiException) {
```

Comment:
Applicable to ALL similar cases in the code.
```suggestion
        } catch (UnzerApiException $unzerApiException) {
```

https://spryker.atlassian.net/wiki/spaces/CORE/pages/955384454/Coding+Guidelines#CodingGuidelines-PHPExceptions

> An exception represents an error. Do not use exceptions in your normal logic flow.